### PR TITLE
[UNDERTOW-2110] Allow line breaks before predicate operator and befor…

### PR DESCRIPTION
**Pull request by baranowb**
…e handler
Issue: https://issues.redhat.com/browse/UNDERTOW-2110

### CODE REVIEW

File: core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java [hunk 1]
```diff
@@ -363,6 +363,11 @@ static Node parse(final String string, Deque<Token> tokens, boolean topLevel) {
                 if(token.getToken().equals(";") && tokens.peek()!=null && tokens.peek().getToken().equals(ELSE)) {
                     // something() -> predicate; ELSE predicate; - dont end processing since its followed by ELSE and its singular block
                     continue;
+                } else if (token.getToken().equals("\n") && tokens.peek() != null && isOperator(tokens.peek().getToken())) {
```
Comment (ropalka) [inline core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java:366]: This doesn't seem correct. What if there are multiple line breaks?

File: core/src/test/java/io/undertow/server/handlers/PredicatedHandlersTestCase.java [hunk 2]
```diff
@@ -159,4 +159,86 @@ public void handleRequest(HttpServerExchange exchange) throws Exception {
...
         Assert.assertEquals("foo'bar",e.getResponseHeaders().get("test-header").getFirst());
     }
+
+    @Test
+    public void testRewriteWithLineBreaks() throws IOException {
+        DefaultServer.setRootHandler(
+                Handlers.predicates(
+
+                        PredicatedHandlersParser.parse(
+                                        "path(/skipallrules)\n and true -> done\n" +
```
Comment (ropalka) [inline core/src/test/java/io/undertow/server/handlers/PredicatedHandlersTestCase.java:169]: Please adjust the test to include multiple line breaks in combination with carriage returns and tabs in a row.

**Review by ropalka — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java (hunk 1 above)
Comment (baranowb) [reply core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java:366]: not allowed. Only allowed at end of handlers section. Even in OG. 
First, tokenization turns /r into /n, secondly,  if you look at if above, it does not allow multiple breaks before ELSE( which is operator).

File: core/src/test/java/io/undertow/server/handlers/PredicatedHandlersTestCase.java (hunk 2 above)
Comment (ropalka) [reply core/src/test/java/io/undertow/server/handlers/PredicatedHandlersTestCase.java:169]: Closing requirement.

File: core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java (hunk 1 above)
Comment (ropalka) [reply core/src/main/java/io/undertow/server/handlers/builder/PredicatedHandlersParser.java:366]: OK

**Review by ropalka — APPROVED** [review]

**Pull request merged**